Monika After Story/game/zz_consumables.rpy

```diff
@@ -1042,12 +1023,23 @@ init 5 python:
             #Verify persist data
             MASConsumable._validatePersistentData(_type)
 
+            if curr_cons:
+                curr_finish_prep_ev = mas_getEV(curr_cons.finish_prep_evl)
+
             #Reset if we're having a consumable we shouldn't be having now and we opened the game after its consume time
+            #or if we're still prepping something but the consumable's finish prepping event doesn't have conditionals
             if (
-                MASConsumable._isHaving(_type)
-                and (
-                    not MASConsumable._isStillCons(_type)
-                    and mas_getCurrSeshStart() > persistent._mas_current_consumable[_type]["consume_time"]
+                (
+                    MASConsumable._isHaving(_type)
+                    and (
+                        not MASConsumable._isStillCons(_type)
+                        and mas_getCurrSeshStart() > persistent._mas_current_consumable[_type]["consume_time"]
+                    )
+                )
+                or (
+                    persistent._mas_current_consumable[_type]["prep_time"] is not None
+                    and curr_finish_prep_ev
+                    and curr_finish_prep_ev.conditional is None
```

Comment:
since this is getting funky, it would be better to create to a static function like `shouldReset`  and just pass in the appropriate parameters from here. Especially since `curr_finish_prep_ev` is only used for this check and not elsewhere in this function.